### PR TITLE
ELEMENTS-1226: vertical alignment consistency with new typography

### PR DESCRIPTION
**Pull request by semisse**

### CODE REVIEW

File: ui/nuxeo-data-table/data-table-row.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ import '@nuxeo/nuxeo-elements/nuxeo-element.js';
             user-select: none;
           }
 
+          :host([header]) {
+            border: none;
```
Comment (Gabez0r) [inline ui/nuxeo-data-table/data-table-row.js:30]: Why do we need this?

File: ui/nuxeo-data-table/data-table-row.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ import '@nuxeo/nuxeo-elements/nuxeo-element.js';
...
+            padding-bottom: 0;
+            border-bottom: none;
```
Comment (Gabez0r) [inline ui/nuxeo-data-table/data-table-row.js:32]: I assume this is not needed because of the `border: none;` above?

File: ui/nuxeo-data-table/default-styles.js [hunk 3]
```diff
@@ -23,9 +23,11 @@ const template = html`
           color: var(--secondary-text-color, #939caa);
           font-weight: 600;
           font-size: 1rem;
-          height: 48px;
-          border-bottom: 2px solid var(--nuxeo-border, #e3e3e3);
-          min-height: 48px;
+          height: 46px;
+          border: none;
```
Comment (Gabez0r) [inline ui/nuxeo-data-table/default-styles.js:27]: Why is this needed?

File: ui/nuxeo-user-group-management/nuxeo-create-group.js [hunk 4]
```diff
@@ -93,7 +93,7 @@ import '../nuxeo-button-styles.js';
 
         <div class="header">
           <iron-icon icon="nuxeo:group"></iron-icon>
-          <h3>[[i18n('createGroup.heading')]]</h3>
+          <h3 class="headline-4">[[i18n('createGroup.heading')]]</h3>
```
Comment (Gabez0r) [inline ui/nuxeo-user-group-management/nuxeo-create-group.js:96]: Should be an `h4` instead?

File: ui/widgets/nuxeo-card.js [hunk 5]
```diff
@@ -107,9 +111,9 @@ import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
 
         <dom-if if="[[_hasHeading(icon, heading, collapsible)]]">
           <template>
-            <h3 class="header" on-click="_toggle">
+            <h3 class="headline-4" on-click="_toggle">
```
Comment (Gabez0r) [inline ui/widgets/nuxeo-card.js:114]: Maybe this should be an `h4` instead?

File: ui/widgets/nuxeo-card.js [hunk 6]
```diff
@@ -93,10 +93,14 @@ import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
 
           .header .heading {
             @apply --layout-flex;
-            font-size: 1rem;
-            font-weight: 700;
-            letter-spacing: 0.04em;
-            text-transform: uppercase;
+            font-weight: 400;
```
Comment (Gabez0r) [inline ui/widgets/nuxeo-card.js:96]: Don thing this `.header .heading` block is being used anymore. Is this intended?

File: ui/nuxeo-data-table/data-table-row.js (hunk 2 above)
Comment (semisse) [reply ui/nuxeo-data-table/data-table-row.js]: Indeed, you are right. I'm changing it to initial to make it clear that we want the default value.

File: ui/nuxeo-data-table/default-styles.js (hunk 3 above)
Comment (semisse) [reply ui/nuxeo-data-table/default-styles.js]: It is not, just an artifact from debugging.

File: storybook/src/elements/widgets/widget.stories.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        display: flex;
+        justify-content: space-between;
+      }
+      .row > * {
+        width: 32%;
+      }
+    </style>
+    <div class="container">
+      <div class="row">
+        <nuxeo-input label="${label}" placeholder="Placeholder"></nuxeo-input>
```
Comment (Gabez0r) [inline storybook/src/elements/widgets/widget.stories.js:33]: nit: a knob for the placeholder could be useful to see how it changes across widgets

**Review by Gabez0r — COMMENTED** [review]
LGTM. Minor misalignment between selectivity multi and single-valued
image, but I can live with it.

File: storybook/.storybook/nuxeo-demo-theme.js [hunk 8]
```diff
@@ -4,11 +4,200 @@ import '@nuxeo/nuxeo-ui-elements/nuxeo-button-styles.js';
...
+
+      h4 {
+        font-weight: 600;
+        font-size: 1rem;
+        letter-spacing: 0;
+        line-height: 1.375rem;
+      }
+
+      h5 {
+        font-weight: 700;
```
Comment (mnixo) [inline storybook/.storybook/nuxeo-demo-theme.js:40]: The `font-weight` for `h4` and `h6` is `600` but for `h5` is `700`?

File: ui/widgets/nuxeo-date-picker.js [hunk 9]
```diff
@@ -144,52 +145,12 @@ import { I18nBehavior } from '../nuxeo-i18n-behavior.js';
...
-          }
-
-          [part='date'][selected][today] {
-            color: var(--nuxeo-button-primary-text) !important;
-          }
-
-          /* hover any date that is not disabled or currently selected */
-          [part='date']:not([disabled]):not([selected]):hover::before {
-            background-color: var(--nuxeo-primary-color) !important;
-            opacity: 0.1 !important;
```
Comment (mnixo) [inline ui/widgets/nuxeo-date-picker.js:192]: This is great!

**Review by mnixo — APPROVED** [review]
👏

**Review by richardsd — DISMISSED** [review]

**Comment by semisse** [comment]
> LGTM. Minor misalignment between selectivity multi and single-valued
> image, but I can live with it.

We will let this improvement for the next topic which will include labels, placeholders, and states consistency.

File: storybook/.storybook/nuxeo-demo-theme.js (hunk 8 above)
Comment (semisse) [reply storybook/.storybook/nuxeo-demo-theme.js:40]: Yes, it's a design decision stated on our [typography guidelines](https://nuxeowiki.atlassian.net/wiki/spaces/DT/pages/889815048/Typography). Thanks for raising this question because I completely forgot to mention this wiki page.

**Comment by richardsd** [comment]
It looks good 👍 Nice work Sam 
vertical_alignment

**Review by mnixo — APPROVED** [review]

**Review by richardsd — DISMISSED** [review]

File: ui/widgets/nuxeo-selectivity.js [hunk 10]
```diff
@@ -7299,6 +7310,14 @@ typedArrayTags[weakMapTag] = false;
 
     _placeholderChanged() {
       this.$.input.setAttribute('placeholder', this.placeholder);
+      const multipleInput = this.shadowRoot.querySelector('.selectivity-multiple-input');
+      if (this.multiple && multipleInput) {
+        multipleInput.setAttribute('placeholder', this.placeholder);
+      }
+      const singleInputPlaceholder = this.shadowRoot.querySelector('.selectivity-placeholder');
```
Comment (Gabez0r) [inline ui/widgets/nuxeo-selectivity.js:7317]: This part should be placed in an else statement, right? There's no point in running this if it is multiple.

**Review by richardsd — DISMISSED** [review]

**Review by Gabez0r — COMMENTED** [review]
I think we forgot `nuxeo-video-info` and `nuxeo-video-conversions`. Both are missing `--nuxeo-label` and the latter also still uses an `h3`. It it's too much work, I'm ok with creating a bug ticket for this.

**Review by mnixo — APPROVED** [review]

**Review by richardsd — APPROVED** [review]
